TLM/TLM/Manager/Impl/SpeedLimitManager.cs

```diff
@@ -397,65 +257,56 @@ or ItemClass.SubService.PublicTransportMetro
             }
 
-            // Resharper warning: condition always false
-            // if (info.name == null) {
-            //    Log._DebugOnlyWarning($"SpeedLimitManager.SetCustomNetInfoSpeedLimitIndex: info.name is null!");
-            //    return -1;
-            // }
-
-            string infoName = info.name;
-            return !customLaneSpeedLimitByNetInfoName_.TryGetValue(infoName, out float speedLimit)
-                       ? GetVanillaNetInfoSpeedLimit(info, true)
+            return !customLaneSpeedLimit_.TryGetValue(info.name, out float speedLimit)
+                       ? GetVanillaNetInfoSpeedLimit(info)
                        : speedLimit;
         }
 
+        internal IEnumerable<NetInfo> GetCustomisableRelatives(NetInfo netinfo) {
+            foreach(var netinfo2 in netinfo.GetRelatives()) {
+                if (customizableNetInfos_.Contains(netinfo2))
+                    yield return netinfo2;
+            }
+        }
+
         /// <summary>
         /// Sets the custom speed limit of the given NetInfo.
         /// </summary>
-        /// <param name="info">the NetInfo for which the custom speed limit should be set</param>
+        /// <param name="netinfo">the NetInfo for which the custom speed limit should be set</param>
         /// <param name="customSpeedLimit">The speed value to set in game speed units</param>
-        public void SetCustomNetInfoSpeedLimit(NetInfo info, float customSpeedLimit) {
-            if (info == null) {
+        public void SetCustomNetinfoSpeedLimit(NetInfo netinfo, float customSpeedLimit) {
+            if (netinfo == null) {
                 Log._DebugOnlyWarning($"SetCustomNetInfoSpeedLimitIndex: info is null!");
                 return;
             }
 
-            string infoName = info.name;
-            customLaneSpeedLimitByNetInfoName_[infoName] = customSpeedLimit;
             float gameSpeedLimit = ToGameSpeedLimit(customSpeedLimit);
 
-            // save speed limit in all NetInfos
-#if DEBUGLOAD
-            Log._Debug($"Updating parent NetInfo {infoName}: Setting speed limit to {gameSpeedLimit}");
-#endif
-            UpdateNetInfoGameSpeedLimit(info, gameSpeedLimit);
 
-            if (childNetInfoNamesByCustomizableNetInfoName_.TryGetValue(
-                infoName,
-                out List<string> childNetInfoNames)) {
-                foreach (string childNetInfoName in childNetInfoNames) {
-                    if (netInfoByName_.TryGetValue(childNetInfoName, out NetInfo childNetInfo)) {
+            foreach (var netinfo2 in GetCustomisableRelatives(netinfo)) {
+                string netinfoName = netinfo2.name;
+                customLaneSpeedLimit_[netinfo2.name] = customSpeedLimit;
+
 #if DEBUGLOAD
-                        Log._Debug($"Updating child NetInfo {childNetInfoName}: Setting speed limit to {gameSpeedLimit}");
+                Log._Debug($"Updating parent NetInfo {netinfoName}: Setting speed limit to {gameSpeedLimit}");
```

Comment:
this is not necessarily parent netinfo. Id remove the parent from this sentence

Reply:
Removed